### PR TITLE
Issue 682: add v2.2.5 improvements for RL to dev

### PR DESCRIPTION
**Pull request by marionbarker**
Apply the patch proposed to solve Issue 682
* https://github.com/ps2/rileylink_ios/issues/682

### CODE REVIEW

**Comment by elnjensen** [comment]
Just to add a data point in support of this PR, I was unable to get my EmaLink to work stably with dev before I applied these changes. Since then, it has worked well.  I don’t know the technical details of what is changed here, but empirically these changes are working for me.

**Comment by ps2** [comment]
This is mostly good, as it's taking updates I applied to master and copying them to dev, but it is backing out some dev changes that are good (crash fixes) that are not in master. I'll try to get a version of this applied to dev soon.

File: RileyLinkBLEKit/Response.swift [hunk 1]
```diff
@@ -103,7 +103,7 @@ struct GetVersionResponse: Response {
     let version: String
 
     init?(data: Data) {
-        guard data.count > 1, let code = ResponseCode(rawValue: data[data.startIndex]) else {
+        guard data.count > 0, let code = ResponseCode(rawValue: data[data.startIndex]) else {
```
Comment (ps2) [inline RileyLinkBLEKit/Response.swift:106]: This, and the following similar changes, are re-introducing crashes.

**Review by ps2 — CHANGES_REQUESTED** [review]

**Comment by marionbarker** [comment]
@ps2 I think you are letting me know that you will make the required changes and that you don't need action from me. Thank you for fixing this.

**Pull request merged**